sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java

```diff
@@ -527,4 +492,57 @@ public void processElement(ProcessContext c) {
       }
     }
   }
+
+  private static <T> Coder<T> getDefaultCreateCoder(CoderRegistry registry, Iterable<T> elems)
+      throws CannotProvideCoderException {
+    // First try to deduce a coder using the types of the elements.
+    Class<?> elementClazz = Void.class;
+    for (T elem : elems) {
+      if (elem == null) {
+        continue;
+      }
+      Class<?> clazz = elem.getClass();
+      if (elementClazz.equals(Void.class)) {
+        elementClazz = clazz;
+      } else if (!elementClazz.equals(clazz)) {
+        // Elements are not the same type, require a user-specified coder.
+        throw new CannotProvideCoderException(
+            String.format(
+                "Cannot provide coder for %s: The elements are not all of the same class.",
+                Create.class.getSimpleName()));
+      }
+    }
+
+    if (elementClazz.getTypeParameters().length == 0) {
+      try {
+        @SuppressWarnings("unchecked") // elementClazz is a wildcard type
+        Coder<T> coder = (Coder<T>) registry.getDefaultCoder(TypeDescriptor.of(elementClazz));
+        return coder;
+      } catch (CannotProvideCoderException exc) {
+        // let the next stage try
```

Comment:
Reword this, maybe something like

"Can't get a coder from the class of the elements, try with the elements next"

or remove (the line on 526 already describes what's happening there)